modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -275,7 +276,17 @@ private void handleTxCleanupCommand(TxCleanupCommand cmd, long commandIndex) {
     /** {@inheritDoc} */
     @Override
     public void onSnapshotSave(Path path, Consumer<Throwable> doneClo) {
-        storage.flush();
+        long maxLastAppliedIndex = Math.max(storage.lastAppliedIndex(), txStateStorage.lastAppliedIndex());
+
+        storage.lastAppliedIndex(maxLastAppliedIndex);
```

Comment:
I think that it would be more correct:
```
        storage.runConsistently(() -> {
            storage.lastAppliedIndex(maxLastAppliedIndex);
            
            return null;
        });
```

Reply:
done